dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -229,15 +231,16 @@ private String getTimeoutMessage(boolean scan) {
 
     private static class TimeoutCheckTask implements TimerTask {
 
-        private DefaultFuture future;
+        private final Long requestID;
 
-        TimeoutCheckTask(DefaultFuture future) {
-            this.future = future;
+        TimeoutCheckTask(Long requestID) {
+            this.requestID = requestID;
         }
 
         @Override
         public void run(Timeout timeout) {
-            if (future == null || future.isDone()) {
+            DefaultFuture future = FUTURES.remove(requestID);
```

Comment:
Personal think that we should use `FUTURES.get` replace `FUTURES.remove`

Reply:
done